### PR TITLE
#164797149 Get articles by Id endpoint 

### PR DESCRIPTION
**Pull request by igbominadeveloper**
#### What does this PR do?
creates an endpoint for paginating articles retrieved from the database

#### Description of Task to be completed?
- [x] write unit tests
- [x] create an endpoint for fetching a single article by its Id
- [x] check and validate the articleId specified is a valid UUID

#### How should this be manually tested?
- Clone the repo,
- Run ```npm install``` to install all dependencies,
- Create .env file using the .env.example as a guide and enter your environment variables
- Run npm test 
- Run npm run start:dev,
- On Postman, send a GET request to ```localhost:3000/api/v1/articles/141f4f05-7d81-4593-ab54-e256c1006210```

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories? (if applicable)
[#164797149](https://www.pivotaltracker.com/story/show/164797149)

#### Screenshots
Screenshot 2019-04-10 at 10 09 43 AM
Screenshot 2019-04-11 at 9 36 05 AM

### CODE REVIEW

File: src/routes/controllers/articles.controller.js [hunk 1]
```diff
@@ -52,3 +52,85 @@ export default async function create(req, res) {
...
 }
+
+/**
+ * Fetch all articles
+ * @param {Object} req Express Request Object
+ * @param {Object} res Express Response Object
+ * @returns {Object} res with articles array if it exists
+ * @returns {Object} res with 404 response if the array is empty
+ */
+export async function getAll(req, res) {
```
Comment (Japhethca) [inline src/routes/controllers/articles.controller.js]: let's remove this controller as it's not related to this ticket.

File: src/routes/users.route.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ *         description: Bad request.
+ *       401:
+ *         description: Authorization information is missing or invalid.
+ *       404:
+ *        description: A user with the specified ID was not found.
+ *       5XX:
+ *        description: Unexpected error.
+ */
+userRouter.post('/', validUser, createUser);
+userRouter.get('/confirmation/:token', confirmUser);
```
Comment (Japhethca) [inline src/routes/users.route.js]: please add the doc for this endpoint

File: src/server.js [hunk 3]
```diff
@@ -56,8 +56,10 @@ app.all('*', (req, res) =>
 );
 
 // finally, let's start our server...
-const server = app.listen(process.env.PORT || 3000, () => {
-  console.log(`Listening on port ${server.address().port}`);
-});
+if (!module.parent) {
```
Comment (Japhethca) [inline src/server.js]: let's remove this check.

File: src/routes/controllers/articles.controller.js (hunk 1 above)
Comment (igbominadeveloper) [reply src/routes/controllers/articles.controller.js]: Okay. I will implement this now

File: src/routes/users.route.js (hunk 2 above)
Comment (igbominadeveloper) [reply src/routes/users.route.js]: Okay. I have added it

File: src/server.js (hunk 3 above)
Comment (igbominadeveloper) [reply src/server.js]: I have removed it

File: src/routes/controllers/articles.controller.js [hunk 4]
```diff
@@ -56,3 +56,42 @@ export default async function create(req, res) {
...
 }
+
+/**
+ * Fetch a single article
+ * @param {Object} req Express Request Object
+ * @param {Object} res Express Response Object
+ * @returns {Object} res with article object if it exists
+ * @returns {Object} res with 404 response if the array is empty
+ */
+export async function findArticle(req, res) {
```
Comment (Japhethca) [inline src/routes/controllers/articles.controller.js]: can we rename this to `getArticle`.

**Review by Japhethca — CHANGES_REQUESTED** [review]
Just few changes

File: src/routes/users.route.js (hunk 2 above)
Comment (igbominadeveloper) [reply src/routes/users.route.js]: This is done now

File: src/routes/controllers/articles.controller.js (hunk 4 above)
Comment (igbominadeveloper) [reply src/routes/controllers/articles.controller.js]: This has been done

**Review by igbominadeveloper — COMMENTED** [review]
I have made the reviews you requested

**Review by Japhethca — APPROVED** [review]

**Pull request merged**